### PR TITLE
Palo Alto Networks Enterprise DLP Integration

### PR DESCRIPTION
**Pull request by megbindal**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
This is the integration for Palo Alto Networks Enterprise DLP. Using this, customers can fetch DLP reports.

## Screenshots
Paste here any images that will help the reviewer
XSOAR-integration

## Minimum version of Demisto
- [x] 5.0.0
- [x] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
...tXcwi+24E+ZiFxx+LF8LdZeb2OL2WDP9+Fyu6BxLpe4V55Xi59gfzxTnkbgKOOZ7f6XxLNRdnuSxTj7L4T3Jph0fqdtSreVnod5hPhTlrh0f6M2brob6dU5v3DfteE/xW4C+x0Kp7AX1z4p3k9vRLXtl+EM7vpKPJ9q7BAFFear4rclbOR4CDqG8ULw33pihSH9TpFzPk2zkd2SEXa5y2RvvrxUp55OJ/5LPIdzfCSX7RpFyP+P+b8WrxPuSkkUf4co9klv5HOZ7X8ARuUdigUTvc4p7TUoWvZdQwgWOXrtcKBOR6P6xqDEqoVTRh/hLiMRjd+ttoUR2jdmHisSWpMPRrTLZe82F4rws5aYrjyNcKIttRb5QvK+JpMOWpCwWyErx7zXtjP3za7x995BfCCWwOOyOU7tP54l83HnemZizjTAHr/NTdgTLDr7YJUYL+dm043kpkVRCbuxV/1ul7c7+h92tDm/8sc/ej/zFflFCJB73gxAJ+iyQW213w0v5EJ9Y3wu4t9F2K2LYkjjdmINi2Hqw23434Q9KiMTjMncigQmBPO3/Ye6RVPI55PedgO4BIz95wkruh4AXiud1TwryZVsQC+Tgs7py35I0ikcg180CsSeqHH2YXc6RLOVzbY7HTVvIkz2W6k/qnhB69DRArrtbdjmDx1bEEMn1CedB7DDv5tRfnGMkFojXg6ZtV2sjXAMLw7YWtmv1xiHeU3KLxD700T7SrVK8H58YLpSo/ykHG3V7CzbsRfHiqytyicQ+EPST7fC692PTjv8Kc/J8+Plm+9WunrAYvtPAKPalGkn4ABYLwjsOY680d0pjVyuVj5V7rWnZv++3yoBnJDH3BlTbrxbC29vf25n0sR5mvVHv2pwZ2e7j50rDnRweLl0iz0hSvzcgCMfFN5qfvSjUSkP474I913g/ideHu5Tm/8JB1xaJnTz6QMAFrikS252wLchG2Df2h8Nm7VoiCYE8FY7hSugjriGSEAjvQzBI6ZEQyPk8PoOySCU/d8t+6OxinY/7/I8oMZJwz/rH4l4ROChpd8viCFuP90QgcFLCliRcAh1u4OcZWsNshINyjSQc07ethV0CTRwYTS6R9E903bXjee8rfPAic0SIJLUzrpvt13BfgP3+GzndH+ColDPVYZd1yN8X8+/Mwg/a7RZdJk91FQAAAABJRU5ErkJggg==
```
Comment (adi88d) [inline Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml:28]:
```suggestion
```
remove this and put the image inside `Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP` folder with the name `Palo_Alto_Networks_Enterprise_DLP_image.png`.
 Logo should match the [requirements](https://xsoar.pan.dev/docs/integrations/integration-logo)(120x50 px PNG, transparent background, Size<10kB )

**Review by adi88d — CHANGES_REQUESTED** [review]
Thanks for your contribution @megbindal! it Looks great.
few notes before:
- Use BaseClient in the code, see this [doc](https://xsoar.pan.dev/docs/integrations/code-conventions#client-class)
- add [function and variables type hints](https://docs.python.org/3/library/typing.html)
- add function doc-strings
- add test playbook
- don't forget to update the README file after changing the yml.

File: Packs/Palo_Alto_Networks_Enterprise_DLP/pack_metadata.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	"vcShouldIgnore": false,
+	"vcShouldKeepItemLegacyProdMachine": false,
+	"commitMessage": "",
+	"shouldCommit": false,
+	"currentVersion": "",
+	"name": "Palo Alto Networks Enterprise DLP Contribution Pack",
+	"description": "Enterprise DLP Integration",
+	"updated": "0001-01-01T00:00:00Z",
+	"created": "2020-11-04T18:13:35Z",
+	"support": "internalContribution",
```
Comment (adi88d) [inline Packs/Palo_Alto_Networks_Enterprise_DLP/pack_metadata.json:19]:
```suggestion
	"support": "xsoar",
```

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    while count < MAX_ATTEMPTS:
+        res = requests.get(url=url, headers={'Authorization': "Bearer " + ACCESS_TOKEN})
+        if res.status_code != 403:
+            break
+        new_token = refreshtoken(ACCESS_TOKEN, REFRESH_TOKEN)
+        if new_token:
+            ACCESS_TOKEN = new_token
+        count += 1
+    if res.status_code < 200 or res.status_code >= 300:
+        raise Exception("Request to {} failed with status code {}".format(url, res.status_code))
```
[the remark this answers is not in the document]
Comment (megbindal) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py]: Done

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+- additionalinfo: Access token generated from Enterprise DLP UI
+  display: Access Token
+  name: access_token
+  required: true
+  type: 4
+- additionalinfo: Refresh token generated from Enterprise DLP UI
+  display: Refresh Token
+  name: refresh_token
+  required: true
+  type: 4
```
Comment (adi88d) [inline Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.yml:15]:
```suggestion
  type: 4
- display: Trust any certificate (not secure)
  name: insecure
  required: false
  type: 8
- display: Use system proxy settings
  name: proxy
  required: false
  type: 8
```

**Review by adi88d — CHANGES_REQUESTED** [review]
please add [function and variables type hints](https://docs.python.org/3/library/typing.html)

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+Palo Alto Networks Enterprise DLP discovers and protect company data across every data channel and repository. Integrated Enterprise DLP enables data protection and compliance everywhere without complexity.
+This integration was integrated and tested with Palo Alto Networks Enterprise DLP
```
[the remark this answers is not in the document]
Comment (megbindal) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/README.md:2]: We don't have a specific version. It's a cloud service and we don't expose the version number

File: Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+                if col not in headers:
+                    headers.append(col)
+                match[col] = detection
+                index += 1
+        matches.append(match)
+    title = 'DLP Report for profile: {}'.format(data_patterns.get("DataProfile"))
+    return tableToMarkdown(title, matches, headers)
+
+
+def makehash():
```
[the remark this answers is not in the document]
Comment (megbindal) [reply Packs/Palo_Alto_Networks_Enterprise_DLP/Integrations/Palo_Alto_Networks_Enterprise_DLP/Palo_Alto_Networks_Enterprise_DLP.py]: Done

**Comment by kirbles19** [comment]
@adi88d Done.

**Review by adi88d — APPROVED** [review]

**Pull request merged**